File: backend/src/main/java/org/globe42/dao/IncomeSourceTypeDao.java
package org.globe42.dao;

import org.globe42.domain.IncomeSourceType;
import org.springframework.data.jpa.repository.JpaRepository;

/**
 * DAO for the {@link IncomeSourceType} entity
 * @author JB Nizet
 */
public interface IncomeSourceTypeDao extends JpaRepository<IncomeSourceType, Long> {
    boolean existsByType(String type);
}

File: backend/src/main/kotlin/org/globe42/dao/IncomeSourceTypeDao.kt
package org.globe42.dao

import org.globe42.domain.IncomeSourceType

/**
 * DAO for the [IncomeSourceType] entity
 * @author JB Nizet
 */
interface IncomeSourceTypeDao : GlobeRepository<IncomeSourceType, Long> {
    fun existsByType(type: String): Boolean
    fun findByType(type: String): IncomeSourceType?
}
